File: task-3/prime.hs
import System.IO

-- Function to check if a number is prime
isPrime :: Int -> Bool
isPrime num
  | num <= 1 = False
  | otherwise = all (\i -> num `mod` i /= 0) [2 .. floor (sqrt $ fromIntegral num)]

-- Main function
main :: IO ()
main = do
  -- Input from the user
  putStr "Enter a number: "
  hFlush stdout
  input <- getLine

  -- Convert input to integer
  let n = read input :: Int

  -- Check if the number is prime
  if isPrime n
    then putStrLn $ show n ++ " is a prime number."
    else putStrLn $ show n ++ " is not a prime number."

File: task-3/prime.java
import java.util.*;

public class Prime {
	int isPrime(int n) {
		if(n<=1) {
			return 0;
		}
		for(int i=2; i<n;i++) {
			if(i%n == 0) {
				return 0;
			}
		}
	return 1;
}

public void main() {
	int n;
	System.out.println("Enter n");
	n = in.nextInt();
	
	if(isPrime(n)) {
		System.out.println(n + "is a Prime no");
	}
	else
	System.out.println(n + " is Not a Prime no");
}
}
